TestWebSocketClient_HeartbeatSucceeds ut flakes（also shown as remotecommand）

### Which jobs are flaking?

https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-unit/1702464725629014016

FAIL: TestWebSocketClient_HeartbeatSucceeds 

k8s.io/kubernetes/vendor/k8s.io/client-go/tools/remotecommand.remotecommand

### Which tests are flaking?

> Failed;  === RUN   TestWebSocketClient_HeartbeatSucceeds
> E0915 00:06:07.556630   64709 websocket.go:478] Websocket Ping failed: write tcp 127.0.0.1:49548->127.0.0.1:38391: use of closed network connection
>     websocket_test.go:1093: unexpected heartbeat timeout
> E0915 00:06:07.606331   64709 websocket.go:478] Websocket Ping failed: write tcp 127.0.0.1:49548->127.0.0.1:38391: use of closed network connection
> --- FAIL: TestWebSocketClient_HeartbeatSucceeds (0.06s)
> 

### Since when has it been flaking?

Added last week in https://github.com/kubernetes/kubernetes/commit/a0d6a815fcc02cbfea1bb22d13a8e896ecbe116c#diff-33da81ac4e24e52fae888fc3a4f2d15ff48fba5b52065fab62bf9f2224cc8204R1044

https://github.com/kubernetes/kubernetes/pull/119157

### Testgrid link

https://testgrid.k8s.io/sig-release-master-blocking#ci-kubernetes-unit

also flake in https://storage.googleapis.com/k8s-triage/index.html?text=HeartbeatSucceeds  ci-kubernetes-unit



### Reason for failure (if possible)

use of closed network connection

### Anything else we need to know?

/cc @seans3 @aojea

### Relevant SIG(s)

/sig api-machinery network cli

--- a/staging/src/k8s.io/client-go/tools/remotecommand/websocket_test.go
+++ b/staging/src/k8s.io/client-go/tools/remotecommand/websocket_test.go
@@ -1057,8 +1057,7 @@ func TestWebSocketClient_HeartbeatSucceeds(t *testing.T) {
 		for {
 			_, _, err := conn.ReadMessage()
 			if err != nil {
-				t.Logf("server err reading message: %v", err)
-				return
+				break
 			}
 		}
 	}))
@@ -1085,7 +1084,11 @@ func TestWebSocketClient_HeartbeatSucceeds(t *testing.T) {
 		return pongHandler(msg)
 	})
 	go heartbeat.start()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
 	go func() {
+		defer wg.Done()
 		for {
 			_, _, err := client.ReadMessage()
 			if err != nil {
@@ -1107,6 +1110,7 @@ func TestWebSocketClient_HeartbeatSucceeds(t *testing.T) {
 		close(heartbeat.closer)
 		t.Errorf("unexpected heartbeat timeout")
 	}
+	wg.Wait()
 }
 
 func TestWebSocketClient_StreamsAndExpectedErrors(t *testing.T) {

